### PR TITLE
Add VSchema DDL support for dropping sequence and auto increment

### PR DESCRIPTION
**Pull request by davidpiegza**
## Description

This PR adds support to drop a `sequence` and `auto_increment` in VSchema in case it needs to be rolled back in a migration.

Drop a sequence
```sql
ALTER VSCHEMA DROP SEQUENCE test_seq
```

Drop auto_increment
```sql
ALTER VSCHEMA ON test_table DROP AUTO_INCREMENT
```

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/13903
Docs: https://github.com/vitessio/website/pull/1575
## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by GuptaManan100** [comment]
@vitessio/query-serving What do you all think of this change?

**Review by deepthi — APPROVED** [review]
LGTM

File: go/vt/topotools/vschema_ddl.go [hunk 1]
```diff
@@ -230,6 +244,21 @@ func ApplyVSchemaDDL(ksName string, ks *vschemapb.Keyspace, alterVschema *sqlpar
...
+
+	case sqlparser.DropAutoIncDDLAction:
+		name := alterVschema.Table.Name.String()
+		table := ks.Tables[name]
+		if table == nil {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "vschema does not contain table %s in keyspace %s", name, ksName)
+		}
+
+		if table.AutoIncrement == nil {
+			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "vschema does not contain auto inc %v on table %s in keyspace %s", table.AutoIncrement, name, ksName)
```
Comment (frouioui) [inline go/vt/topotools/vschema_ddl.go:257]: nit picking:

```suggestion
			return nil, vterrors.Errorf(vtrpcpb.Code_INTERNAL, "vschema does not contain auto increment %v on table %s in keyspace %s", table.AutoIncrement, name, ksName)
```

**Review by frouioui — APPROVED** [review]
Thanks for this first contribution @davidpiegza! Everything looks good to me outside of that small nit.

I see there is a handful of workflows that are stuck in `expected`, can you push another commit to re-trigger the CI please.

**Comment by davidpiegza** [comment]
@frouioui @deepthi Is there anything left I can do or is it ready to merge?

**Pull request merged**